### PR TITLE
ROC-2905: Add single claimant and single defendant requirement for eligibility

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-2905

### Change description
Added 2 pages to the eligibility journey, inserted into the 2nd and 3rd position.
These pages will make sure a single claim can not have multiple claimants and a single claim cannot have multiple defendants.

### Corresponding Integration Tests ###
https://github.com/hmcts/cmc-integration-tests/pull/63

### Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

File: src/main/features/claim/helpers/eligibility/notEligibleReason.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ export class NotEligibleReason {
   static readonly CLAIM_VALUE_NOT_KNOWN: string = 'claim-value-not-known'
   static readonly CLAIM_VALUE_OVER_10000: string = 'claim-value-over-10000'
   static readonly OVER_18: string = 'over-18'
+  static readonly MULTIPLE_CLAIMANTS: string = 'multiple-claimant'
```
Comment (geek-soft) [inline src/main/features/claim/helpers/eligibility/notEligibleReason.ts]: Isn't `multiple-claimants` more correct? The same line below.

File: src/main/features/claim/views/eligibility/not-eligible.njk [hunk 2]
```diff
@@ -7,6 +7,7 @@
 {% block content %}
 
   {% set n1FormUrl = 'https://formfinder.hmctsformfinder.justice.gov.uk/n1-eng.pdf' %}
+  {% set moneyClaimOnline = 'https://www.moneyclaim.gov.uk/web/mcol/welcome' %}
```
Comment (geek-soft) [inline src/main/features/claim/views/eligibility/not-eligible.njk]: What the `moneyClaimOnline` is? Name is not meaningful enough for me. What about `legacyServiceUrl`

File: src/main/features/claim/views/eligibility/not-eligible.njk [hunk 3]
```diff
@@ -20,6 +21,15 @@
           + '<a target="_blank" href="%s">form N1</a> ' + newWindow() + '.',
           { postProcess: 'sprintf', sprintf: [ n1FormUrl ]}) | safe }}
         </p>
+      {% elseif reason === NotEligibleReason.MULTIPLE_CLAIMANTS %}
+        <p>{{ t('You can’t use this service if more than one person or business is making the claim') }}</p>
+        <p>{{ t('Use <a target="_blank" href="%s">form N1 </a>' + newWindow() + ' to make your claim..',
+          { postProcess: 'sprintf', sprintf: [ n1FormUrl ]}) | safe }}</p>
```
Comment (geek-soft) [inline src/main/features/claim/views/eligibility/not-eligible.njk]: Do not use `sprintf` processor anymore, use interpolation provided by i18next See: 

https://www.i18next.com/interpolation.html

File: src/main/features/claim/views/eligibility/single-claimant.njk [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <h1 class="heading-large">{{ t('Are you the only person or business making the claim?') }}</h1>
+      <p>{{ t('You can’t use this service if more than one person or business is making the claim') }}</p>
+
+      {% set name = 'singleClaimant' %}
+
+      <form novalidate method="post" class="analytics-submit-event-trigger"
+            data-event-action="Eligibility: Select Yes"
```
Comment (geek-soft) [inline src/main/features/claim/views/eligibility/single-claimant.njk]: This is not specific enough. A `Eligibility: Select Yes` is sent to GA so that we can see what people select. Seeing `Eligibility: Select Yes` only does not help us understand what Yes / No values mean. I would suggest something in line with `Eligibility: Single claimant`. Having this we will see the following in the GA:

```
Eligibility: Single claimant | Yes | 34
Eligibility: Single claimant | No  | 12
```

File: src/main/features/claim/views/eligibility/single-claimant.njk [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            data-event-label-from="{{ name }}">
+        {{ csrfProtection(csrf) }}
+
+        {{
+          radioGroup(
+            'Are you the only person or business making the claim?',
```
Comment (geek-soft) [inline src/main/features/claim/views/eligibility/single-claimant.njk]: Please implicitly name argument names for readability e.g.:

```
radioGroup(
  label = 'Are you the only person or business making the claim?',
  ...
```

File: src/main/features/claim/views/eligibility/single-defendant.njk [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      {% set name = 'singleDefendant' %}
+
+      <form novalidate method="post" class="analytics-submit-event-trigger"
+            data-event-action="Eligibility: Select Yes"
+            data-event-label-from="{{ name }}">
+        {{ csrfProtection(csrf) }}
+
+        {{
+          radioGroup(
+            'Is this claim against only one individual or business?',
```
Comment (geek-soft) [inline src/main/features/claim/views/eligibility/single-defendant.njk]: The same here please.

File: src/main/features/claim/views/eligibility/single-defendant.njk [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            name,
+            [
+              {'label': 'Yes', value: YesNoOption.YES.option, checked: (form.valueFor(name) === YesNoOption.YES)},
+              {'label': 'No', value: YesNoOption.NO.option, checked: (form.valueFor(name) === YesNoOption.NO)}
+            ],
+            form = form,
+            labelForScreenReadersOnly = true
+          )
+        }}
+
```
Comment (geek-soft) [inline src/main/features/claim/views/eligibility/single-defendant.njk]: No need to have two blank lines.

File: src/main/features/claim/views/eligibility/single-claimant.njk [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            name,
+            [
+              {'label': 'Yes', value: YesNoOption.YES.option, checked: (form.valueFor(name) === YesNoOption.YES)},
+              {'label': 'No', value: YesNoOption.NO.option, checked: (form.valueFor(name) === YesNoOption.NO)}
+            ],
+            form = form,
+            labelForScreenReadersOnly = true
+          )
+        }}
+
```
Comment (geek-soft) [inline src/main/features/claim/views/eligibility/single-claimant.njk]: No need for two blank lines.

File: src/test/app/drafts/models/deserialization.ts [hunk 9]
```diff
@@ -84,6 +84,7 @@ describe('DraftClaim deserialization', () => {
     expect(deserialized.eligibility.claimantAddress).to.equal(YesNoOption.YES)
     expect(deserialized.eligibility.defendantAddress).to.equal(YesNoOption.YES)
     expect(deserialized.eligibility.eighteenOrOver).to.equal(YesNoOption.YES)
+    expect(deserialized.eligibility.singleClaimant).to.equal(YesNoOption.YES)
```
Comment (geek-soft) [inline src/test/app/drafts/models/deserialization.ts]: Isn't worth doing the same for `singleDefendant` property?

File: src/test/features/claim/form/models/eligibility/eligibility.ts [hunk 10]
```diff
@@ -14,14 +14,7 @@ describe('Eligibility', () => {
...
       const errors = validator.validateSync(
-        new Eligibility(
-          YesNoOption.YES,
-          YesNoOption.YES,
-          ClaimValue.UNDER_10000,
-          YesNoOption.YES,
-          YesNoOption.NO,
-          undefined
-        )
+        new Eligibility(YesNoOption.YES, YesNoOption.YES, ClaimValue.UNDER_10000, YesNoOption.NO, YesNoOption.YES, undefined)
```
Comment (geek-soft) [inline src/test/features/claim/form/models/eligibility/eligibility.ts]: Is that still working? You made constructor accept 8 args while you provide only 6.

File: src/main/features/claim/form/models/eligibility/eligibility.ts [hunk 11]
```diff
@@ -46,10 +56,11 @@ export class Eligibility {
       YesNoOption.fromObject(input.claimantAddress),
       YesNoOption.fromObject(input.defendantAddress),
       ClaimValue.fromObject(input.claimValue),
+      YesNoOption.fromObject(input.singleClaimant),
+      YesNoOption.fromObject(input.singleDefendant),
       YesNoOption.fromObject(input.eighteenOrOver),
       YesNoOption.fromObject(input.governmentDepartment),
-      YesNoOption.fromObject(input.helpWithFees)
-    )
+      YesNoOption.fromObject(input.helpWithFees))
```
Comment (timja-kainos) [inline src/main/features/claim/form/models/eligibility/eligibility.ts]: undo this change please

File: src/main/features/claim/views/eligibility/not-eligible.njk [hunk 12]
```diff
@@ -20,6 +21,15 @@
...
+      {% elseif reason === NotEligibleReason.MULTIPLE_DEFENDANTS %}
+        <p>{{ t('Use <a target="_blank" href="%s">Money Claim Online </a>' + newWindow() + ', if you’re claiming against two people.',
+          { postProcess: 'sprintf', sprintf: [ moneyClaimOnline ]}) | safe }}</p>
+        <p>{{ t('If your claim is against more than two individuals or companies, use <a target="_blank" href="%s">form N1 </a>' + newWindow() + ' to make your claim.',
+          { postProcess: 'sprintf', sprintf: [ n1FormUrl ]}) | safe }}</p>
```
Comment (timja-kainos) [inline src/main/features/claim/views/eligibility/not-eligible.njk]: on line 38 of this file it also goes to mcol, please update now that it is being used in more than one place to use the variable
REQ

File: src/main/features/claim/views/eligibility/single-defendant.njk [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <h1 class="heading-large">{{ t('Is this claim against only one individual or business?') }}</h1>
+      <p>{{ t('You can’t use this service if this claim is against more than one person or business.') }}</p>
+
+      {% set name = 'singleDefendant' %}
+
+      <form novalidate method="post" class="analytics-submit-event-trigger"
+            data-event-action="Eligibility: Select Yes"
```
Comment (timja-kainos) [inline src/main/features/claim/views/eligibility/single-defendant.njk]: same GA comment as above from damian

File: src/test/features/claim/form/models/eligibility/eligibility.ts (hunk 10 above)
Comment (timja-kainos) [reply src/test/features/claim/form/models/eligibility/eligibility.ts]: well it's not complete so it will still pass, the values are optional in the constructor so it will compile,
it's not as complete as before and it's harder to read, 
please revert to old style and add the extra options
REQ

File: src/test/features/claim/form/models/eligibility/eligibility.ts [hunk 14]
```diff
@@ -30,14 +23,7 @@ describe('Eligibility', () => {
...
       const errors = validator.validateSync(
-        new Eligibility(
-          YesNoOption.YES,
-          YesNoOption.YES,
-          ClaimValue.UNDER_10000,
-          YesNoOption.YES,
-          YesNoOption.NO,
-          YesNoOption.NO
-        )
+        new Eligibility(YesNoOption.YES, YesNoOption.YES, ClaimValue.UNDER_10000, YesNoOption.NO, YesNoOption.YES, YesNoOption.NO)
```
Comment (timja-kainos) [inline src/test/features/claim/form/models/eligibility/eligibility.ts]: same here

File: src/main/features/claim/views/eligibility/not-eligible.njk (hunk 2 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/not-eligible.njk]: In that sense, the n1FormUrl isnt meaningful enough either. 
I think legacyServiceUrl is a good recommendation. Ill add now

Comment (timja-kainos) [reply src/main/features/claim/views/eligibility/not-eligible.njk]: huh, how is it not meaningful?
I would be fine with leaving how it is.

Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/not-eligible.njk]: Unless you know exactly what the N1 form does, its no more helpful than a "m84FormUrl"

File: src/main/features/claim/views/eligibility/single-claimant.njk (hunk 4 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/single-claimant.njk]: Great explanation.
Ill add what you suggested

File: src/main/features/claim/views/eligibility/single-claimant.njk (hunk 5 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/single-claimant.njk]: I was working off the Over_18 file which didn't have that. 
I will add it in now

File: src/main/features/claim/views/eligibility/single-claimant.njk (hunk 8 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/single-claimant.njk]: Working off the over_18 page which had them. Ive updated them all to remove the extra line

File: src/test/app/drafts/models/deserialization.ts (hunk 9 above)
Comment (jacmurphy) [reply src/test/app/drafts/models/deserialization.ts]: Done

File: src/test/features/claim/form/models/eligibility/eligibility.ts (hunk 10 above)
Comment (jacmurphy) [reply src/test/features/claim/form/models/eligibility/eligibility.ts]: Done

File: src/main/features/claim/views/eligibility/single-claimant.njk (hunk 8 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/single-claimant.njk]: Removed but this comment will not update as it covers 1 extra line

File: src/main/features/claim/views/eligibility/single-defendant.njk (hunk 7 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/single-defendant.njk]: Removed but this comment will not update as it covers 1 extra line

File: src/main/features/claim/views/eligibility/not-eligible.njk (hunk 12 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/not-eligible.njk]: Done.

File: src/main/features/claim/views/eligibility/not-eligible.njk (hunk 3 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/not-eligible.njk]: As all the other reasons use sprintf, would you like me to rework them too?

Comment (timja-kainos) [reply src/main/features/claim/views/eligibility/not-eligible.njk]: separate PR

File: src/main/features/claim/views/eligibility/over-18.njk [hunk 15]
```diff
@@ -19,7 +19,7 @@
 
         {{
           radioGroup(
-            'Are you 18 or over?',
+            label 'Are you 18 or over?',
```
Comment (timja-kainos) [inline src/main/features/claim/views/eligibility/over-18.njk]: `label = `

Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/over-18.njk]: Done

File: src/main/features/claim/views/eligibility/not-eligible.njk (hunk 3 above)
Comment (jacmurphy) [reply src/main/features/claim/views/eligibility/not-eligible.njk]: 👍
